packages/context/src/inject.ts

```diff
@@ -188,7 +188,9 @@ export function inject(
 /**
  * The function injected by `@inject.getter(bindingSelector)`.
  */
-export type Getter<T> = () => Promise<T>;
+export interface Getter<T> {
+  (): Promise<T>;
+}
```

Comment:
What is the reasoning for this change? Do you need an interface so that you can inherit from it?